File: archive/c/c-sharp/Base64EncodeDecode.cs
using System.Text;

return args switch
{
    ["encode", var value] when !string.IsNullOrWhiteSpace(value)
        => Encode(value),

    ["decode", var value] when !string.IsNullOrWhiteSpace(value)
        => Decode(value),

    _ => Usage()
};

static int Encode(string value)
{
    Console.WriteLine(Convert.ToBase64String(Encoding.ASCII.GetBytes(value)));
    return 0;
}

static int Decode(string value)
{
    byte[] buffer = new byte[value.Length];
    if (!Convert.TryFromBase64String(value, buffer, out int written))
        return Usage();

    Console.WriteLine(Encoding.ASCII.GetString(buffer, 0, written));
    return 0;
}

static int Usage()
{
    Console.Error.WriteLine("Usage: please provide a mode and a string to encode/decode");
    return 1;
}

File: archive/v/visual-basic/base64-encode-decode.vb
Imports System.Text

Module Program

    Sub Main(args As String())

        If args.Length <> 2 Then
            ShowUsage()
            Return
        End If

        Dim mode = args(0).ToLowerInvariant()
        Dim value = args(1)

        If String.IsNullOrWhiteSpace(mode) OrElse String.IsNullOrWhiteSpace(value) Then
            ShowUsage()
            Return
        End If

        Select Case mode

            Case "encode"
                Console.WriteLine(
                    Convert.ToBase64String(
                        Encoding.UTF8.GetBytes(value)
                    )
                )

            Case "decode"
                Try
                    Console.WriteLine(
                        Encoding.UTF8.GetString(
                            Convert.FromBase64String(value)
                        )
                    )

                Catch ex As FormatException
                    ShowUsage()
                End Try

            Case Else
                ShowUsage()

        End Select

    End Sub

    Private Sub ShowUsage()
        Console.WriteLine("Usage: please provide a mode and a string to encode/decode")
    End Sub

End Module